### ISSUE TITLE
Swift 2.3 support

### ISSUE THREAD

**Issue by mz2** [OAuthSwift/OAuthSwift#239]
There are a few API changes in Swift 2.3 (or rather, in nullability annotations of Foundation) which make OAuthSwift not compile cleanly with Swift 2.3.

I've run the syntax migration tool for my fork over here: https://github.com/mpapp/OAuthSwift – this is simplistic though as it (probably?) breaks Swift 2.2 and earlier given the nullability changes.

**Comment by phimage** [comment]
I see your change on `absoluteString`(now optional), thx

Swift 2.3 is available as part of Xcode 8, not yet officially released, so I can modify the master branch
I can just create a new branch from the latest stable release and commit the change

**Comment by mz2** [comment]
There's a preprocessor statement available for that as of Swift 2.2, which would let you support both: 

```
#if swift(>=2.3)
…
```

**Comment by phimage** [comment]
Ok the solution could be into NSURL+OAuthSwift.swift

```swift
var unsafeAbsoluteString: String {
     #if swift(>=2.3)

}

**Comment by phimage** [comment]
os updated, xcode beta launched, and fix for 2.3 commited

**Comment by mz2** [comment]
Sweet, thanks!
